nixos/modules/system/boot/stage-2.nix

```diff
@@ -78,12 +73,10 @@ in
       };
 
       systemdExecutable = mkOption {
-        default = "systemd";
+        default = "/run/current-system/systemd/lib/systemd/systemd";
         type = types.str;
         description = ''
-          The program to execute to start systemd. Typically
-          <literal>systemd</literal>, which will find systemd in the
-          PATH.
+          The program to execute to start systemd.
```

Comment:
Why does this stay configurable, but now defaults to `/run/current-system/systemd/lib/systemd/`?

This smells a bit like we have some semi-static interface / assumptions now here? The default (and why) should probably be documented in the description.

Reply:
You can ask @grahamc about that: #93110. It won't work with the new systemd stage 1 anymore anyway because we need to transition to systemd directly

Reply:
We default it to /run now because I wanted to drop $PATH